src/ScriptTransform.ts

```diff
@@ -0,0 +1,14 @@
+import * as esbuild from "esbuild-wasm";
+import wasmURL from "../node_modules/esbuild-wasm/esbuild.wasm";
+
+let resolved = false; //this is ugly but lets us skip awaiting a resolved promise
+const initPromise = esbuild
+  .initialize({
+    wasmURL,
+  })
+  .then(() => (resolved = true));
+
+export const transform: typeof esbuild.transform = async (input, options) => {
+  if (!resolved) await initPromise;
+  return esbuild.transform(input, options);
+};
```

Comment:
Transform is already a promise-based API, so what's the big deal here?

Why not just

```suggestion
const initPromise = esbuild.initialize({wasmURL});

export const transform: typeof esbuild.transform = (input, options) => {
  return initPromise.then(_ => esbuild.transform(input, options));
};
```